### PR TITLE
 On branch data-structure_stack

### PR DESCRIPTION
**Pull request by ryoasaka12345**
Changes to be committed:
	new file:   stack/hw1.php
	new file:   stack/stack.php

### CODE REVIEW

File: DataStructure/stack/hw1.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        return count($this->elements);
+    }
+}
+
+function hw1($words)
+{
+    $stack = new Stack_hw1();
+
+    for ($i=0; $i < count($words); $i++) {
```
Comment (pav-thanhtt) [inline DataStructure/stack/hw1.php]: add space between `$i=0` like below
`for ($i = 0; $i < count($words); $i++) {`

by the way, for better performance update like below
```
$countWords = count($words);
for ($i = 0; $i < $countWords; $i++) {
```

File: DataStructure/stack/hw1.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        $top = $stack->top();
+        if ($top == $words[$i] ) {
```
Comment (pav-thanhtt) [inline DataStructure/stack/hw1.php]: use `===` instead of `==`

File: DataStructure/stack/hw1.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+require ('stack.php');
+
+class Stack_hw1 extends Stack
```
Comment (pa-vuhn) [inline DataStructure/stack/hw1.php]: classname should be PascalCase

File: DataStructure/stack/stack.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    * @return int
+    */
+    public function top()
+    {
+           if (!$this->isEmpty()) {
+            return $this->elements[sizeof($this->elements) - 1];
+        }
+    
+        return null;    
+    }
```
Comment (pav-thanhtt) [inline DataStructure/stack/stack.php]: indent 4 spaces

**Pull request merged**